### ISSUE TITLE
Accuracy of model reproduction

### ISSUE THREAD

**Issue by shibing15** [rsy6318/CorrI2P#11]
Dear Siyu,  
Thanks again for sharing your great job CorrI2P, I generated KITTI data according to DeepI2P and the way you mentioned without splicing multi-frame point cloud together, the point cloud number is 20480, and carried out experimental reproduction. After 25 epoches, training loss dropped from 4.78 to 3.34 and descriptor loss dropped from 3.95 to 3.17. The evaluation accuracy of the model is RTE 3.6402 + -49.7405 RRE 5.6616 + -19.2482. I am not sure if there are some problems with my dataset processing. It is too difficult to download the dataset you provided, and even if the download succeeds, many files are lost. The data processing script you uploaded looks no different from deepi2p's dataset , there is also multi-frame concatenation operation, so could you share your code for preprocessing Kitti? In addition, I see that the name of your point cloud file has been normalized. What does this normalization mean?  I am looking forward to your reply.

**Comment by rsy6318** [comment]
see https://github.com/rsy6318/CorrI2P/tree/main/kitti_script to prepare kitti dataset.

**Comment by rsy6318** [comment]
How do you evaluate the accuracy?

**Comment by shibing15** [comment]
The model accuracy is evaluated according to the eval_all.py, cal_error_all.py and cal_metry. py scripts that you uploaded

**Comment by shibing15** [comment]
> Did you visual your prepared pointcloud?
Yeah, I visualized some point cloud data, like this
![image](https://github.com/rsy6318/CorrI2P/assets/29483573/84be51bb-5afd-4bb4-942a-6f2743ac37d9)

**Comment by rsy6318** [comment]
I mean that you should project point cloud on the image using gt and predicted transformation.

**Comment by shibing15** [comment]
> I mean that you should project point cloud on the image using gt and predicted transformation.
I found that the accuracy evaluation standard mentioned in the paper only included the data with  t_error less than **5m** and angle_error less than **10 &deg;**, which was different from the evaluation method used by cal_metric.py. After modification, it reached RTE 1.0008 + -0.7571, RRE 2.8308 + -2.0258. 
Thank you very much for your patience!

**Comment by shibing15** [comment]
> > see https://github.com/rsy6318/CorrI2P/tree/main/kitti_script to prepare kitti dataset.
 
Yes, this script is used to generate the data set without concatenating the multi-frame point cloud as you mentioned.
> > [#2 (comment)](https://github.com/rsy6318/CorrI2P/issues/2#issuecomment-1236080480)
